Exporter exiting when no data is returned  from a metric definition available

Exporter exiting when no data exists for  one metric.

```2018/11/12 13:11:00 GET https://management.azure.com/subscriptions/XXXXXXXXXXX/resourceGroups/XXXXXXXsbn-rg/providers/Microsoft.ServiceBus/namespaces/XXXXXXXX/providers/microsoft.insights/metrics?aggregation=Total%2CAverage%2CMinimum%2CMaximum&api-version=2018-01-01&metricnames=SuccessfulRequests%2CServerErrors%2CUserErrors%2CIncomingRequests%2CIncomingMessages%2COutgoingMessages%2CActiveConnections%2CSize%2CMessages%2CActiveMessages%2CDeadletteredMessages%2CScheduledMessages%2CThrottledRequests&timespan=2018-11-12T13%3A07%3A00Z%2F2018-11-12T13%3A08%3A00Z
panic: runtime error: index out of range

goroutine 16 [running]:
panic(0x3d3640, 0xc42000a0f0)
	/usr/local/Cellar/go/1.7.5/libexec/src/runtime/panic.go:500 +0x1a1
main.(*Collector).Collect(0x678d90, 0xc420327500)
	/usr/local/Cellar/go/1.7.5/bin/src/github.com/RobustPerception/azure_metrics_exporter/main.go:71 +0xd97
github.com/RobustPerception/azure_metrics_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func2(0xc420396ad0, 0xc420327500, 0x63b720, 0x678d90)
	/usr/local/Cellar/go/1.7.5/bin/src/github.com/RobustPerception/azure_metrics_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:383 +0x63
created by github.com/RobustPerception/azure_metrics_exporter/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).Gather
	/usr/local/Cellar/go/1.7.5/bin/src/github.com/RobustPerception/azure_metrics_exporter/vendor/github.com/prometheus/client_golang/prometheus/registry.go:384 +0x326
```

Config:

``` targets:
  - resource: "/resourceGroups/xXXXX-rg/providers/Microsoft.ServiceBus/namespaces/XXXX"
    metrics:
      - name: "SuccessfulRequests"
      - name: "ServerErrors"
      - name: "UserErrors"
      - name: "IncomingRequests"
      - name: "IncomingMessages"
      - name: "OutgoingMessages"
      - name: "ActiveConnections"
      - name: "Size"
      - name: "Messages"
      - name: "ActiveMessages"
      - name: "DeadletteredMessages"
      - name: "ScheduledMessages"
      - name: "ThrottledRequests"
```

ThrottledRequests -> This is the guilty guy. If configured only on its own or removed from the list. there is no failure.


```╰─ ./azure_metrics_exporter --list.definitions
2018/11/12 12:55:14 Resource: XXXXX

Available Metrics:
2018/11/12 12:55:14 - SuccessfulRequests
2018/11/12 12:55:14 - ServerErrors
2018/11/12 12:55:14 - UserErrors
2018/11/12 12:55:14 - ThrottledRequests
2018/11/12 12:55:14 - IncomingRequests
2018/11/12 12:55:14 - IncomingMessages
2018/11/12 12:55:14 - OutgoingMessages
2018/11/12 12:55:14 - ActiveConnections
2018/11/12 12:55:14 - Size
2018/11/12 12:55:14 - Messages
2018/11/12 12:55:14 - ActiveMessages
2018/11/12 12:55:14 - DeadletteredMessages
2018/11/12 12:55:14 - ScheduledMessages```

--- a/main.go
+++ b/main.go
@@ -72,39 +72,42 @@ func (c *Collector) extractMetrics(ch chan<- prometheus.Metric, rm resourceMeta,
 		metricName = strings.ToLower(metricName + "_" + value.Unit)
 		metricName = strings.Replace(metricName, "/", "_per_", -1)
 		metricName = invalidMetricChars.ReplaceAllString(metricName, "_")
-		metricValue := value.Timeseries[0].Data[len(value.Timeseries[0].Data)-1]
-		labels := CreateResourceLabels(rm.resourceURL)
 
-		if hasAggregation(rm.aggregations, "Total") {
-			ch <- prometheus.MustNewConstMetric(
-				prometheus.NewDesc(metricName+"_total", metricName+"_total", nil, labels),
-				prometheus.GaugeValue,
-				metricValue.Total,
-			)
-		}
+		if len(value.Timeseries) > 0 {
+			metricValue := value.Timeseries[0].Data[len(value.Timeseries[0].Data)-1]
+			labels := CreateResourceLabels(rm.resourceURL)
 
-		if hasAggregation(rm.aggregations, "Average") {
-			ch <- prometheus.MustNewConstMetric(
-				prometheus.NewDesc(metricName+"_average", metricName+"_average", nil, labels),
-				prometheus.GaugeValue,
-				metricValue.Average,
-			)
-		}
+			if hasAggregation(rm.aggregations, "Total") {
+				ch <- prometheus.MustNewConstMetric(
+					prometheus.NewDesc(metricName+"_total", metricName+"_total", nil, labels),
+					prometheus.GaugeValue,
+					metricValue.Total,
+				)
+			}
 
-		if hasAggregation(rm.aggregations, "Minimum") {
-			ch <- prometheus.MustNewConstMetric(
-				prometheus.NewDesc(metricName+"_min", metricName+"_min", nil, labels),
-				prometheus.GaugeValue,
-				metricValue.Minimum,
-			)
-		}
+			if hasAggregation(rm.aggregations, "Average") {
+				ch <- prometheus.MustNewConstMetric(
+					prometheus.NewDesc(metricName+"_average", metricName+"_average", nil, labels),
+					prometheus.GaugeValue,
+					metricValue.Average,
+				)
+			}
 
-		if hasAggregation(rm.aggregations, "Maximum") {
-			ch <- prometheus.MustNewConstMetric(
-				prometheus.NewDesc(metricName+"_max", metricName+"_max", nil, labels),
-				prometheus.GaugeValue,
-				metricValue.Maximum,
-			)
+			if hasAggregation(rm.aggregations, "Minimum") {
+				ch <- prometheus.MustNewConstMetric(
+					prometheus.NewDesc(metricName+"_min", metricName+"_min", nil, labels),
+					prometheus.GaugeValue,
+					metricValue.Minimum,
+				)
+			}
+
+			if hasAggregation(rm.aggregations, "Maximum") {
+				ch <- prometheus.MustNewConstMetric(
+					prometheus.NewDesc(metricName+"_max", metricName+"_max", nil, labels),
+					prometheus.GaugeValue,
+					metricValue.Maximum,
+				)
+			}
 		}
 	}
 

